handler.go

```diff
@@ -172,7 +173,12 @@ func (h *Handler) ServeHTTP(responseWriter http.ResponseWriter, request *http.Re
 	}
 
 	// Find our implementation of the RPC protocol in use.
-	contentType := request.Header.Get("Content-Type")
+	encodedContentType := request.Header.Get("Content-Type")
+	contentTypeList := strings.Split(encodedContentType, ";")
+	var contentType = encodedContentType
+	if len(contentTypeList) > 1 {
+		contentType = strings.TrimSpace(contentTypeList[0])
```

Comment:
This might be a good place to use https://pkg.go.dev/mime#ParseMediaType instead.